builder/vsphere/common/step_add_cdrom.go

```diff
@@ -81,10 +82,15 @@ func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multiste
 	vm := state.Get("vm").(driver.VirtualMachine)
 
 	if s.Config.CdromType == "sata" {
-		if _, err := vm.FindSATAController(); err == driver.ErrNoSataController {
-			ui.Say("Adding SATA controller...")
-			if err := vm.AddSATAController(); err != nil {
-				state.Put("error", fmt.Errorf("error adding SATA controller: %v", err))
+		if _, err := vm.FindSATAController(); err != nil {
+			if errors.Is(err, driver.ErrNoSataController) {
```

Comment:
Very nitpicky, but I would suggest inverting this one to reduce the nesting by one level for the `ErrNoSataController` case :)

Reply:
Updated in 9e0a0d3.